From a59830e64116234751bcdf165a5df4badbef5b8d Mon Sep 17 00:00:00 2001
From: Myron Stowe <myron.stowe@redhat.com>
Date: Thu, 10 Jul 2014 01:02:31 -0400
Subject: [acpi] hotplug/pci: Drop redundant checks from check_hotplug_bridge()

Message-id: <20140710010230.22146.94168.stgit@gir.stowe>
Patchwork-id: 85580
O-Subject: [RHEL7.1 PATCH 041/105] ACPI / hotplug / PCI: Drop redundant checks from check_hotplug_bridge()
Bugzilla: 1114228
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>

Bugzilla: https://bugzilla.redhat.com/show_bug.cgi?id=1114228
Upstream Status: 07bb735378919e4b5863077f5c1b4037b6ca1a99

commit 07bb735378919e4b5863077f5c1b4037b6ca1a99
Author: Rafael J. Wysocki <rafael.j.wysocki@intel.com>
Date:   Sat Jul 13 23:27:25 2013 +0200

    ACPI / hotplug / PCI: Drop redundant checks from check_hotplug_bridge()

    Two checks in check_hotplug_bridge() are redundant (they have been
    done by the caller already), so drop them.

    Signed-off-by: Rafael J. Wysocki <rafael.j.wysocki@intel.com>
    Tested-by: Mika Westerberg <mika.westerberg@linux.intel.com>
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/pci/hotplug/acpiphp_glue.c b/drivers/pci/hotplug/acpiphp_glue.c
index 7ec569d..ddd08a5 100644
--- a/drivers/pci/hotplug/acpiphp_glue.c
+++ b/drivers/pci/hotplug/acpiphp_glue.c
@@ -586,16 +586,10 @@ static void check_hotplug_bridge(struct acpiphp_slot *slot, struct pci_dev *dev)
 {
  struct acpiphp_func *func;
 
- if (!dev->subordinate)
-  return;
-
  /* quirk, or pcie could set it already */
  if (dev->is_hotplug_bridge)
   return;
 
- if (PCI_SLOT(dev->devfn) != slot->device)
-  return;
-
  list_for_each_entry(func, &slot->funcs, sibling) {
   if (PCI_FUNC(dev->devfn) == func->function) {
    dev->is_hotplug_bridge = 1;
-- 
1.7.1